jupyterhub/handlers/base.py

```diff
@@ -402,6 +414,7 @@ def finish_user_spawn(f=None):
                 spawner.add_poll_callback(self.user_stopped, user)
             finally:
                 spawner._proxy_pending = False
+                self.hub.spawn_pending_count -= 1
```

Comment:
I think we should factor stuff out of basehandler at some point - using settings for this feels icky, but is probably the right thing to do for now.